packages/proxy/lib/http/response-middleware.ts

```diff
@@ -345,6 +345,27 @@ const OmitProblematicHeaders: ResponseMiddleware = function () {
   this.next()
 }
 
+const MaybeSetOriginAgentClusterHeader: ResponseMiddleware = function () {
+  if (process.env.CYPRESS_INTERNAL_E2E_TESTING_SELF_PARENT_PROJECT) {
+    const origin = new URL(this.req.proxiedUrl).origin
+
+    if (process.env.HTTP_PROXY_TARGET_FOR_ORIGIN_REQUESTS && process.env.HTTP_PROXY_TARGET_FOR_ORIGIN_REQUESTS === origin) {
+      // For cypress-in-cypress tests exclusively, we need to bucket all origin-agent-cluster requests
+      // from HTTP_PROXY_TARGET_FOR_ORIGIN_REQUESTS to include Origin-Agent-Cluster=false. This has to due with changed
+      // behavior starting in Chrome 119. The new behavior works like the following:
+      //     If a page did not request an origin-keyed agent cluster, chrome will place it in one
+      //     anyway because a previous request went through without the Origin-Agent-Cluster=false header set.
+      //     At time of writing, documentation detailing this behavior has not been found.
```

Comment:
I believe the documentation is [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin-Agent-Cluster):

> The browser will ensure that all pages from a given origin are either origin-keyed or they are not. This means that:
>
> * If the first page from an origin does not set the header, then no other pages from that origin will be origin-keyed, even if those other pages do set the header.
> * If the first page from an origin sets the header and is made origin-keyed, then all other pages from that origin will be origin-keyed whether they ask for it or not.